From d8177a8f3ba6cc6e80d27a7c6bdd1b5920e12544 Mon Sep 17 00:00:00 2001
From: Giovanni Campagna <gcampagna@src.gnome.org>
Date: Mon, 12 Aug 2013 13:14:37 +0000
Subject: ScreenShield: don't allow events through the lock dialog

Make the lock dialog group reactive, to intercept any events
before they go to the actors below.
In the future, we may restructure our chrome to have a clear
layer system, but for now it fixes a security issue in the lock
screen (you can see the contents of the windows by dragging
if the screen was locked with the overview active)

https://bugzilla.gnome.org/show_bug.cgi?id=705840
---
diff --git a/js/ui/screenShield.js b/js/ui/screenShield.js
index 4264f01..8452bc6 100644
--- a/js/ui/screenShield.js
+++ b/js/ui/screenShield.js
@@ -478,6 +478,7 @@ const ScreenShield = new Lang.Class({
 
         this._lockDialogGroup = new St.Widget({ x_expand: true,
                                                 y_expand: true,
+                                                reactive: true,
                                                 opacity: 0,
                                                 pivot_point: new Clutter.Point({ x: 0.5, y: 0.5 }),
                                                 name: 'lockDialogGroup' });
--
cgit v0.9.2
